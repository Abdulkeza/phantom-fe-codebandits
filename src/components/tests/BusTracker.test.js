import { render } from '@testing-library/react';
import React from 'react';
import { MapContainer } from 'react-leaflet';
import BusTracker from '../BusTracker.js';

const data = [
  {
    lat: -1223,
    lng: 30.0
  },
  {
    lat: -12223,
    lng: 31.0
  }
];

describe('BusTracker Page', () => {
  it('Should start bus tracking', async () => {
    const elem = render(
      <MapContainer
        center={{ lat: -1.936671, lng: 30.053524 }}
        zoom={13}
        zoomControl={false}
        className="h-screen md:h-[88vh] w-[95vw]"
      >
        <BusTracker data={data[0]} />
      </MapContainer>
    );

    expect(elem).toMatchSnapshot();
  });
});

