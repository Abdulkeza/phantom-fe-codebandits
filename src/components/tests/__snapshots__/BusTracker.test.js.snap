// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusTracker Page Should start bus tracking 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen md:h-[88vh] w-[95vw] leaflet-container leaflet-grab leaflet-touch-drag"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="busicon.png"
              style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -311px; top: 1037292px; z-index: 1037292; transform-origin: center; transform:  rotateZ(360deg);"
              tabindex="0"
            />
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          />
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen md:h-[88vh] w-[95vw] leaflet-container leaflet-grab leaflet-touch-drag"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        />
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <img
            alt=""
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            src="busicon.png"
            style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -311px; top: 1037292px; z-index: 1037292; transform-origin: center; transform:  rotateZ(360deg);"
            tabindex="0"
          />
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        />
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
